src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -497,21 +490,29 @@ public byte[] Http2AltSvcOriginUri
             Socket? socket = null;
             SslStream? sslStream = null;
             TransportContext? transportContext = null;
+            IReadOnlyList<Http2Connection>? lastHttp2Connections = _http2Connections;
```

Comment:
Why `IReadOnlyList<Http2Connection>` rather than `Http2Connection[]`?  The former is much less efficient.

Reply:
It's a leftover from some previous version. Will fix it.